File: 4.js
var groupAnagrams = function (strs) {
	let map = {};
	for (let i = 0; i < strs.length; i++) {
		let str = strs[i];
		let key = str.split("").sort().join("");
		if (map[key] === undefined) {
			map[key] = [str];
		} else {
			map[key].push(str);
		}
	}
	let result = [];
	for (let key in map) {
		result.push(map[key]);
	}
	return result;
};

File: 4.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
  vector<vector<string>> groupAnagrams(vector<string> &strs)
  {
    vector<vector<string>> res;
    unordered_map<string, vector<string>> mp;
    for (auto &s : strs)
    {
      string temp = s;
      sort(temp.begin(), temp.end());
      mp[temp].push_back(s);
    }

    for (auto &m : mp)
    {
      res.push_back(m.second);
    }
    return res;
  }
};